Themis/Views/Assessment/CodeEditor/CodeView.swift

```diff
@@ -1,130 +1,113 @@
+    var body: some View {
+        ZStack {
+            CodeEditor(source: $file.code ?? "loading...",
+                       language: .swift,
+                       theme: theme,
+                       fontSize: $fontSize,
+                       flags: editorFlags,
+                       highlightedRanges: cvm.inlineHighlights[file.path] ?? [],
+                       dragSelection: $dragSelection,
+                       line: $line,
+                       showAddFeedback: $cvm.showAddFeedback,
+                       selectedSection: $cvm.selectedSection)
+            if let line {
+                DrawingShape(points: line.points)
+                    .stroke(line.color, style: StrokeStyle(lineWidth: line.lineWidth, lineCap: .round, lineJoin: .round))
             }
-        }
-    }
-    var file: Node? {
-        didSet {
-            if textView.text != file?.code {
-                applySyntaxHighlighting(on: textView)
+        }.gesture(DragGesture(minimumDistance: 0, coordinateSpace: .local).onChanged({ value in
+            let newPoint = value.location
+            if value.translation.width + value.translation.height == 0 {
+                self.line = Line(
+                    points: [newPoint],
+                    color: .orange.opacity(0.5),
+                    lineWidth: 10.0
+                )
+            } else {
+                self.line?.points.append(newPoint)
             }
-        }
+        }).onEnded { _ in
+            if let dragSelection {
+                onOpenFeedback(dragSelection)
+            }
+            dragSelection = nil
+            line = nil
+        })
     }
 
-    init(cvm: CodeEditorViewModel) {
-        self.cvm = cvm
-        super.init(nibName: nil, bundle: nil)
+    var editorFlags: CodeEditor.Flags {
+        if colorScheme == .dark {
+            return [.selectable, .blackBackground]
+        } else {
+            return .selectable
+        }
     }
 
-    required init?(coder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
+    var theme: CodeEditor.ThemeName {
+        if colorScheme == .dark {
+            return .ocean
+        } else {
+            return .xcode
+        }
     }
+}
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        navigationController?.navigationBar.scrollEdgeAppearance = UINavigationBarAppearance()
-        textView.translatesAutoresizingMaskIntoConstraints = false
-        setCustomization(on: textView)
-        view.addSubview(textView)
-        NSLayoutConstraint.activate([
-            textView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
-            textView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
-            textView.topAnchor.constraint(equalTo: view.topAnchor),
-            textView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
-        ])
-        setupLongPressInteraction()
-    }
+/// used for testing editmenu
+ struct CodeViewSelect: View {
```

Comment:
We should remove this by implementing the lasso toggle functionality. But merging is more important for now.